multidim-interop/src/compose-runner.ts

```diff
@@ -0,0 +1,59 @@
+import { tmpdir } from 'os'
+import { promises as fs } from 'fs';
+import path from 'path';
+import { exec as execStd } from 'child_process';
+import util from 'util';
+import { env } from 'process';
+import { ComposeSpecification, PropertiesServices } from "../compose-spec/compose-spec"
+import { stringify } from 'yaml';
+
+const exec = util.promisify(execStd);
+
+export type RunOpts = {
+    up: {
+        exitCodeFrom: string
+        renewAnonVolumes?: boolean
+    }
+}
+
+export type RunFailure = any
+
+export async function run(namespace: string, compose: ComposeSpecification, opts: RunOpts): Promise<RunFailure | null> {
+    // sanitize namespace
+    namespace = namespace.replace(/[^a-zA-Z0-9]/g, "-")
+    const dir = path.join(tmpdir(), "compose-runner", namespace)
```

Comment:
If I understand correctly `tmpdir()` only returns the OS' default tmp directory. Is there any function to create a new temporary directory, like `os.MkdirTemp` in Go? Then you could also remove the check if the directory exists below.

Reply:
There might be, but this is nice because you make sure to clean up old things in case of a crash that didn't get time to cleanup.